src/entities/inputValidator/gyro/inputValidatorGyro.ts

```diff
@@ -15,11 +13,127 @@ import {
+            throw new Error(
+                `StretchingFactorWrong: lambda must be <= ${_MAX_STRETCH_FACTOR}`,
+            );
+        }
+
+        // validateDerivedParamsLimits
+        const { derivedEclpParams } = input;
+        const { tauAlpha, tauBeta, dSq, u, v, w, z } = derivedEclpParams;
+
+        // require(derived.tauAlpha.y > 0, DerivedTauAlphaYWrong());
+        if (tauAlpha.y <= 0n)
+            throw new Error('DerivedTauAlphaYWrong: tuaAlpha.y must be > 0');
+
+        // require(derived.tauBeta.y > 0, DerivedTauBetaYWrong());
+        if (tauBeta.y <= 0n)
+            throw new Error('DerivedTauBetaYWrong: tauBeta.y must be > 0');
+
+        // require(derived.tauBeta.x > derived.tauAlpha.x, DerivedTauXWrong());
+        if (tauBeta.x <= tauAlpha.x)
+            throw new Error(
+                'DerivedTauXWrong: derived.tauBeta.x must be > derived.tauAlpha.x',
+            );
+
+        // require(_ONE_XP - _DERIVED_TAU_NORM_ACCURACY_XP <= norm2 && norm2 <= _ONE_XP + _DERIVED_TAU_NORM_ACCURACY_XP,DerivedTauAlphaNotNormalized());
+        const norm2 = GyroECLPMath.scalarProdXp(tauAlpha, tauAlpha);
+        if (norm2 < MIN_NORM2)
+            throw new Error(
+                `DerivedTauBetaNotNormalized: norm2 must be >= ${MIN_NORM2}`,
+            );
+        if (norm2 > MAX_NORM2)
+            throw new Error(
+                `DerivedTauBetaNotNormalized: norm2 must be <= ${MAX_NORM2}`,
+            );
+
+        // require(derived.u <= _ONE_XP, DerivedUWrong());
+        if (u > _ONE_XP)
+            throw new Error(`DerivedUWrong: u must be <= ${_ONE_XP}`);
+        // require(derived.v <= _ONE_XP, DerivedVWrong());
+        if (v > _ONE_XP)
+            throw new Error(`DerivedVWrong: v must be <= ${_ONE_XP}`);
+        // require(derived.w <= _ONE_XP, DerivedWWrong());
+        if (w > _ONE_XP)
+            throw new Error(`DerivedWWrong: w must be <= ${_ONE_XP}`);
+        // require(derived.z <= _ONE_XP, DerivedZWrong());
+        if (z > _ONE_XP)
+            throw new Error(`DerivedZWrong: z must be <= ${_ONE_XP}`);
+
+        // require(_ONE_XP - _DERIVED_DSQ_NORM_ACCURACY_XP <= derived.dSq && derived.dSq <= _ONE_XP + _DERIVED_DSQ_NORM_ACCURACY_XP, DerivedDsqWrong());
+        if (dSq < MIN_DSQ)
+            throw new Error(`DerivedDSqWrong: dSq must be >= ${MIN_DSQ}`);
+        if (dSq > MAX_DSQ)
+            throw new Error(`DerivedDSqWrong: dSq must be <= ${MAX_DSQ}`);
+
+        // require(mulDenominator <= _MAX_INV_INVARIANT_DENOMINATOR_XP, InvariantDenominatorWrong());
+        const achiachi = GyroECLPMath.calcAChiAChiInXp(
+            eclpParams,
+            derivedEclpParams,
+        );
+        const mulDenominator = (_ONE_XP * _ONE_XP) / achiachi;
+        if (mulDenominator > _MAX_INV_INVARIANT_DENOMINATOR_XP)
+            throw new Error(
+                `InvariantDenominatorWrong: mulDenominator must be <= ${_MAX_INV_INVARIANT_DENOMINATOR_XP}`,
+            );
```

Comment:
All the ECLP param validations

https://github.com/gyrostable/gyro-pools/blob/50f21c3ae61bd84375006f985c05438482627631/contracts/eclp/GyroECLPMath.sol#L85-L128